### PR TITLE
feat: add coin_selection module with token-aware UTXO selection

### PR DESCRIPTION
**Pull request by fpelliccioni**
## Summary
- Replace legacy `select_outputs` (unused libbitcoin inheritance, no token/fee awareness) with modern `coin_selection` module in `wallet/`
- Three functions: `select_utxos`, `select_utxos_send_all`, `select_utxos_both`
- Two strategies: `clean` (default, wallet-friendly) and `mixed` (experimental, reports collateral tokens)
- Fixes token-burning bug: clean mode skips token-carrying UTXOs when selecting for BCH
- 20 test cases covering all modes and edge cases

## Test plan
- [ ] Verify native build compiles cleanly
- [ ] Run coin_selection tests
- [ ] Verify WASM build is not affected

---

> [!NOTE]
> **Medium Risk**
> Touches wallet coin-selection and transaction-template construction, including token collateral handling and dust/fee calculations; mistakes could create invalid transactions or risk token loss. Changes are well-covered by extensive new unit tests but still affect core spend-building paths.
> 
> **Overview**
> Adds a new `wallet/coin_selection` module that centralizes UTXO selection and unsigned transaction template creation with **CashTokens awareness**, including clean vs mixed selection (with collateral-token reporting), send-all flows, and a `create_token_split_tx_template` to split “dirty” token+excess-BCH UTXOs.
> 
> Moves transaction template creation out of `chain::transaction` into `wallet::create_tx_template`, updates the C API to call the new wallet API, and switches coin-selection enum conversions from `domain::chain` to `domain::wallet`.
> 
> Removes legacy `wallet/select_outputs` and the old `chain/coin_selection` header, updates build/export headers accordingly, and adds a large `coin_selection` test suite covering fee/dust boundaries, change distribution correctness, and token-collateral edge cases.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit d788231a3eb8e2da94f24a9972a1a7a6a29e6836. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

## Summary by CodeRabbit

* **New Features**
  * Unified coin/UTXO selection and transaction-template capabilities for BCH and tokens: clean vs mixed selection, send-all, token-splitting, improved fee/size estimation, and flexible change distribution.

* **Tests**
  * Extensive unit tests covering BCH, fungible-token, NFT selection, mixed flows, send-all, token-split templates, and fee/dust edge cases.

* **Chores**
  * Replaced legacy output-selection with the new coin-selection API and updated C API/transaction template callers.

### CODE REVIEW

File: src/domain/src/wallet/coin_selection.cpp [hunk 1]
```diff
@@ -0,0 +1,715 @@
...
+            output_addresses.push_back(change_address);
+            output_amounts.push_back(token_dust_limit);
+        }
+
+        for (auto const& token : sel.collateral_non_fungible_tokens) {
+            outputs.push_back(make_token_output(token_dust_limit, change_address, token));
+            output_addresses.push_back(change_address);
+            output_amounts.push_back(token_dust_limit);
+        }
+    }
```
[the remark this answers is not in the document]
Comment (fpelliccioni) [reply src/domain/src/wallet/coin_selection.cpp:746]: Tracked in https://github.com/k-nuth/kth/issues/206. Currently unreachable: `create_tx_template` always selects BCH-only in `clean` mode (both `bch_id` and `coin_selection_strategy::clean` are now explicit in the call). The code path for collateral outputs exists for future token sending support (https://github.com/k-nuth/kth/issues/207), at which point the BCH deduction will be implemented.

File: src/domain/src/wallet/coin_selection.cpp [hunk 2]
```diff
@@ -0,0 +1,701 @@
...
+// -- Fee estimation --
+
+uint64_t estimate_fee(size_t num_inputs, size_t outputs_size) {
+    auto const size = base_tx_size + (num_inputs * approx_input_size) + outputs_size;
+    return size * sats_per_byte;
+}
+
+// -- Transaction building helpers --
+
+constexpr uint32_t tx_version = 2;
```
[the remark this answers is not in the document]
Comment (fpelliccioni) [reply src/domain/src/wallet/coin_selection.cpp:111]: Not a bug. Version 2 aligns with BCHN's default (`CTransaction::CURRENT_VERSION = 2`). All inputs use max_input_sequence (`0xFFFFFFFF`) which sets `SEQUENCE_LOCKTIME_DISABLE_FLAG`, disabling `BIP68` relative lock-time. With max sequence on all inputs, version 1 and 2 are functionally identical. The old code used version 1 (matching Electron Cash's default), but version 2 is the modern standard per BCHN.

File: src/domain/src/wallet/coin_selection.cpp [hunk 3]
```diff
@@ -0,0 +1,769 @@
...
+        std::sort(available_utxos.begin(), available_utxos.end(),
+            [](chain::utxo const& a, chain::utxo const& b) { return a.amount() < b.amount(); });
+    } else if (selection_algo == coin_selection_algorithm::largest_first) {
+        std::sort(available_utxos.begin(), available_utxos.end(),
+            [](chain::utxo const& a, chain::utxo const& b) { return a.amount() > b.amount(); });
+    }
+
+    // In send_all mode, only the destination output is created (no change).
+    size_t const output_count = send_all ? 1 : change_addresses.size() + 1;
+    size_t const outputs_size = output_count * approx_output_size;
```
[the remark this answers is not in the document]
Comment (fpelliccioni) [reply src/domain/src/wallet/coin_selection.cpp:654]: Tracked in https://github.com/k-nuth/kth/issues/209. This requires an iterative fee estimation approach (estimate → distribute → filter dust → re-estimate with fewer outputs → repeat until stable).

File: src/domain/test/wallet/coin_selection.cpp [hunk 4]
```diff
@@ -0,0 +1,1305 @@
...
+    // 4 outputs: 1 dest + 3 change
+    REQUIRE(amounts.size() == 4);
+    REQUIRE(amounts[0] == 10000);  // destination
+    // Change amounts: first two get truncated value, last gets remainder
+    // total_change = 12000 - 10000 - 294 = 1706
+    // amounts[1] = uint64_t(1706 * 1/3) = 568
+    // amounts[2] = uint64_t(1706 * 1/3) = 568
+    // amounts[3] = 1706 - 568 - 568 = 570 (remainder)
+    REQUIRE(amounts[1] + amounts[2] + amounts[3] == 1706);
+}
```
[the remark this answers is not in the document]
Comment (fpelliccioni) [reply src/domain/test/wallet/coin_selection.cpp:1000]: This is the same underlying issue reported in https://github.com/k-nuth/kth/pull/201#discussion_r2994752857 
Tracked in https://github.com/k-nuth/kth/issues/209. 
The regression test will be added together with the iterative fee re-estimation fix.

File: src/domain/src/wallet/coin_selection.cpp [hunk 5]
```diff
@@ -0,0 +1,774 @@
...
```
[the remark this answers is not in the document]
Comment (fpelliccioni) [reply src/domain/src/wallet/coin_selection.cpp:654]: Same root cause as https://github.com/k-nuth/kth/pull/201#discussion_r2994752857. 
Tracked in https://github.com/k-nuth/kth/issues/209 — the fix requires iterative fee re-estimation until the output count stabilizes.
